packages/core/src/App/Containers/RealAccountSignup/financial-details.jsx

```diff
@@ -0,0 +1,862 @@
+import {
+    AutoHeightWrapper,
+    DesktopWrapper,
+    Div100vhContainer,
+    FormSubmitButton,
+    MobileWrapper,
+    Dropdown,
+    ThemedScrollbars,
+    SelectNative,
+} from '@deriv/components';
+import { Formik, Field } from 'formik';
+import React from 'react';
+import { FormSubHeader } from '@deriv/account';
+import { localize, Localize } from '@deriv/translations';
+import { isDesktop, isMobile } from '@deriv/shared/utils/screen';
+import { connect } from 'Stores/connect';
+
+class FinancialDetails extends React.Component {
+    form = React.createRef();
+    async componentDidMount() {
+        await this.form.current.getFormikActions().validateForm();
+    }
+
+    componentWillUnmount() {
+        this.is_mounted = false;
+    }
+
+    handleCancel = values => {
+        this.props.onSave(this.props.index, values);
+        this.props.onCancel();
+    };
+
+    render() {
+        const padding_bottom = window.innerHeight < 930 ? '10rem' : '12rem';
+        return (
+            <Formik
+                initialValues={{ ...this.props.value }}
+                validate={this.props.validate}
+                onSubmit={(values, actions) => {
+                    this.props.onSubmit(this.props.index, values, actions.setSubmitting);
+                }}
+                ref={this.form}
+            >
+                {({ handleSubmit, isSubmitting, errors, values, setFieldValue, handleChange, handleBlur, touched }) => (
+                    <AutoHeightWrapper default_height={200}>
+                        {({ setRef, height }) => (
+                            <form ref={setRef} onSubmit={handleSubmit}>
+                                <Div100vhContainer
+                                    className='details-form'
+                                    height_offset='199px'
+                                    is_disabled={isDesktop()}
+                                >
+                                    <p className='details-form__description'>
+                                        <Localize i18n_default_text="We're legally obliged to ask for your financial information." />
+                                    </p>
+                                    <ThemedScrollbars
+                                        is_native={isMobile()}
+                                        autoHide={!(window.innerHeight < 890)}
+                                        height={height}
+                                    >
+                                        <div
```

Comment:
Is there any way we can break these form components into other components in the same file? No logic change, just for ease of digestion.

Reply:
Consider it done.